app/javascript/components/Lobby.js

```diff
@@ -9,27 +10,51 @@ const service_directory = {
   ktv: "🎤 ",
 }
 
-const Lobby = (props) => (
-  <Layout position={props.position}>
-    { Object.keys(service_directory).map((service_name) => (
-      <Service key={service_name}>
-        <Emoji>{ service_directory[service_name] }</Emoji>
-
-        <Tables>
-          {props.services.filter(s => s.service == service_name).map((table) => (
-            <Table
-              current_time={props.current_time}
-              key={table.position}
-              refresh={props.refresh}
-              service={service_name}
-              {...table}
-            />
-          ))}
-        </Tables>
-      </Service>
-    ))}
-  </Layout>
-)
+class Lobby extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.timer = null
+
+    this.state = { current_time: moment() }
+  }
+
+  componentDidMount() {
+    this.timer = setInterval(() => this.setTime(), 1000)
+  }
+
+  setTime() {
+    this.setState({ current_time: moment() })
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timer)
+  }
+
+  render() {
+    return (
+      <Layout position={this.props.position}>
+        { Object.keys(service_directory).map((service_name) => (
+          <Service key={service_name}>
+            <Emoji>{ service_directory[service_name] }</Emoji>
```

Comment:
did you make a custom html tag?  or what is this?

Reply:
It's a React component - essentially a custom HTML tag that has javascript behaviors and styling built-in.

https://reactjs.org/

![thefuture](https://media.giphy.com/media/e7i6KJU8jBMvS/giphy.gif)

Reply:
Also see https://www.styled-components.com/ for the styling part of it.